source/containers.tex

```diff
@@ -72,12 +72,10 @@
 In Tables~\ref{tab:containers.container.requirements},
 \ref{tab:containers.reversible.requirements}, and
 \ref{tab:containers.optional.operations}
-\tcode{X} denotes a container class containing objects of type
-\tcode{T}, \tcode{a} and \tcode{b}
-denote values of type \tcode{X}, \tcode{u}
```

Comment:
How confident are you that `a` and `b` here weren't supposed to be possibly-const? The distinction between them and `r` makes me worry that (eg) the `r = a` requirement below was supposed to apply for both the case where `a` is of type `X` and the case where `a` is of type `const X`.

Reply:
The expressions `a = rv` and `a.swap(b)` require non-const, so `a` and `b` are definitely not allowed to be const in those rows. However, you have revealed an existing bug: most of the expressions _should_be using possibly-const values, e.g. `X(a)` and `X u(a); X u = a;` (yes I know the latter aren't expressions :smiley:) and `(&a)->~X()`, and `a.begin()` and `a.end()`, `a.cbegin()` and `a.cend()`, `a == b` and `a != b`, `a.size()` etc. etc. etc. etc.

So I think we should make `a` and `b` possibly const, and introduce new variables for use in the handful of rows that aren't valid for const objects.

Reply:
I'll create an LWG issue for that.

Reply:
https://wg21.link/lwg3028